nodes/stampzilla-server/protocol/nodes.go

```diff
@@ -64,14 +65,24 @@ func (n *node) SetConn(conn net.Conn) {
 	n.Host = conn.RemoteAddr().String()
 	n.Unlock()
 }
-func (n *node) Write(b []byte) {
+func (n *node) Write(b []byte) error {
 	b = append(b, []byte("\n")...)
 	_, err := n.conn.Write(b)
 	if err != nil {
 		n.conn.Close()
 		log.Error(err)
-		return
+		return err
 	}
+
+	return nil
+}
+func (n *node) WriteUpdate(msg *protocol.Update) error {
```

Comment:
maybe WriteUpdate does not need to be a method. It could take the node as first argument with typ io.Writer